core/lib/zksync_core/src/consistency_checker/mod.rs

```diff
@@ -1,204 +1,300 @@
         let commit_tx_id = storage_l1_batch
             .eth_commit_tx_id
-            .unwrap_or_else(|| panic!("Commit tx not found for L1 batch #{}", batch_number))
+            .with_context(|| format!("Commit tx not found for L1 batch #{batch_number}"))?
             as u32;
-
         let block_metadata = storage
             .blocks_dal()
             .get_l1_batch_with_metadata(storage_l1_batch)
-            .await
-            .unwrap()
-            .unwrap_or_else(|| {
-                panic!(
-                    "Metadata for L1 batch #{} not found in the database",
-                    batch_number
-                )
-            });
-
+            .await?
+            .with_context(|| {
+                format!("Metadata for L1 batch #{batch_number} not found in the database")
+            })?;
         let commit_tx_hash = storage
             .eth_sender_dal()
             .get_confirmed_tx_hash_by_eth_tx_id(commit_tx_id)
-            .await
-            .unwrap()
-            .unwrap_or_else(|| {
-                panic!(
-                    "Commit tx hash not found in the database. Commit tx id: {}",
-                    commit_tx_id
-                )
-            });
-
-        tracing::info!(
-            "Checking commit tx {} for batch {}",
-            commit_tx_hash,
-            batch_number.0
-        );
-
-        // we can't get tx calldata from db because it can be fake
-        let commit_tx = self
-            .web3
-            .eth()
-            .transaction(TransactionId::Hash(commit_tx_hash))
             .await?
-            .expect("Commit tx not found on L1");
+            .with_context(|| {
+                format!("Commit tx hash not found in the database for tx id {commit_tx_id}")
+            })?;
+        drop(storage);
+
+        tracing::info!("Checking commit tx {commit_tx_hash} for L1 batch #{batch_number}");
 
         let commit_tx_status = self
-            .web3
-            .eth()
-            .transaction_receipt(commit_tx_hash)
+            .l1_client
+            .transaction_status(commit_tx_hash)
             .await?
-            .expect("Commit tx receipt not found on L1")
-            .status;
+            .with_context(|| format!("Receipt for tx {commit_tx_hash:?} not found on L1"))?;
+        if commit_tx_status != 1.into() {
+            let err = anyhow::anyhow!("Main node gave us a failed commit tx");
+            return Err(err.into());
+        }
 
-        assert_eq!(
-            commit_tx_status,
-            Some(1.into()),
-            "Main node gave us a failed commit tx"
-        );
+        // We can't get tx calldata from db because it can be fake.
+        let commit_tx_input_data = self
+            .l1_client
+            .transaction_input_data(commit_tx_hash)
+            .await?
+            .with_context(|| format!("Commit for tx {commit_tx_hash:?} not found on L1"))?;
+        // FIXME: shouldn't the receiving contract and selector be checked as well?
```

Comment:
I'm not very familiar with the checker requirements, but the fact that it doesn't check either the recipient contract address, or the Solidity function selector makes it intuitively easy to fool.

Reply:
Good catch, agree. I'd prefer it to be implemented separately though.

Reply:
Added a TODO.